k--- .orig/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
+++ /usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
@@ -70,8 +70,14 @@
 			
 			// Create a filename using the image count and path info stored in preferences.
 			var imgNum = this.prefs[CameraControl.PREFS.COUNT];
-			var imgNumStr = "000"+imgNum;
-			imgNumStr = imgNumStr.substring(imgNumStr.length-4);  
+			var date = new Date();
+			var YYYY = date.getFullYear();
+			var MM = (1+date.getMonth()); if (MM<10) { MM = "0"+MM; }
+			var DD = (0+date.getDate()); if (DD<10) { DD = "0"+DD; }
+			var hh = date.getHours(); if (hh < 10) { hh = "0"+hh; }
+			var mm = date.getMinutes(); if (mm < 10) { mm = "0"+mm; }
+			var ss = date.getSeconds(); if (ss < 10) { ss = "0"+ss; }
+			var fileBase = YYYY + "-" +  MM + "-" + DD + "-" + hh + mm + ss;
 	
 			// Set file destination image
 			cam.dst = this.prefs[CameraControl.PREFS.PATH]
@@ -79,25 +85,11 @@
 								+ this.prefs[CameraControl.PREFS.MAKE].toUpperCase().substring(0,4)
 								+ "/"
 								+ this.prefs[CameraControl.PREFS.FILE]
-								+ imgNumStr
+								+ fileBase
 								+".jpg";
 
 			// Increment image count, but wrap at 10000, then save back to preferences
 			++imgNum;
-			if (imgNum > 9999){
-				imgNum = 1;
-	
-				// Increment the album number (directory) so we don't overwrite the previous CIMG0001.jpg 
-				var album = (+this.prefs[CameraControl.PREFS.ALBUM]);
-				if (album == 900) {
-					album = 100;
-				}
-				else {
-					++album;
-				}
-				CameraPrefs.updatePref(this.prefs, CameraControl.PREFS.ALBUM, album);
-			}
-	
 			CameraPrefs.updatePref(this.prefs, CameraControl.PREFS.COUNT, imgNum);
 		}
 
@@ -215,19 +207,6 @@
 		
 		// Increment image count - after all, the image we got from the adapter was the LAST CAPTURED
 		++image;
-		if (image > 9999) {
-			image = 1;
-			
-			// Increment the album number (directory) so we don't overwrite the previous CIMG0001.jpg 
-			if (album == 900) {
-				album = 100;
-			}
-			else {
-				++album;
-			}
-			CameraPrefs.updatePref(this.prefs, CameraControl.PREFS.ALBUM, album);
-		}
-		
 		CameraPrefs.updatePref(this.prefs, CameraControl.PREFS.COUNT, image);
 		
 		llog("ALBUM: " + album);
--- .orig/usr/palm/applications/com.palm.app.camera/camera-prefs-defaults.json
+++ /usr/palm/applications/com.palm.app.camera/camera-prefs-defaults.json
@@ -7,7 +7,7 @@
 	'CAMERA-MAKE': 'Palm',
 	'CAMERA-MODEL': 'Pre ',
 	'COUNT': '1',
-	'FILE-PREFIX': 'CIMG',
+	'FILE-PREFIX': 'PRE_',
 	'GEOTAGGING': 'enabled',
 	'GPS-ACCURACY-METER': 'disabled',
 	'GPS-FIX-MAX-AGE': '120',

